Src/FluentAssertions/Equivalency/Steps/GenericDictionaryEquivalencyStep.cs

```diff
@@ -114,11 +114,6 @@ private static bool AssertExpectationIsNotNull(object subject, object expectatio
         // the AssertSameLength method.
         where TExpectedKey : TSubjectKey
     {
-        if (expectation.Count == subject.Count)
```

Comment:
Why do you think that this is redundant?

Reply:
#935 added an earlier `Count` check on line 105, but didn't remove this one.

So I'm not sure this inner one can be hit anymore.

Reply:
I had a closer look at this and I'm good with this change.